schema.c

```diff
@@ -3890,6 +3890,19 @@ _sch_traverse_nodes (sch_node * schema, GNode * parent, int flags, int depth, in
                 name = NULL;
             }
         }
+        else if (!child && (flags & SCH_F_SET_NULL) && g_strcmp0 (name, "*") == 0)
```

Comment:
This should be in the sch_is_leaf_list section below.
Also please use the fn's for detecting leaf-lists rather than comp name and casting to xmlNode